### PR TITLE
Lvjv 30

### PR DESCRIPTION
**Pull request by maryshchur**
Added user entities

### CODE REVIEW

**Review by ArturSydor — DISMISSED** [review]

**Review by ArturSydor — DISMISSED** [review]

File: src/main/java/com/softserve/rms/entities/User.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    @Column(nullable = false)
+    private String status;
+//TODO mapping on roles table
+//    @ManyToOne (fetch = FetchType.LAZY)
+//    @JoinColumn(name = "role_id")
+//    private Role role;
+
+    @OneToMany(mappedBy = "user", orphanRemoval = true)
+    private List<ResourceTemplate> resource_templates = new ArrayList<>();
```
Comment (halinayatseniuk) [inline src/main/java/com/softserve/rms/entities/User.java]: It's better to use resourceTemplates field name.
Why do you initialize an empty list of resource templates for a user? What should have a user with admin or registrator role in this list? Including the fact that admin should have access to users and registrator to resources only.

**Review by MarianDutchyn — DISMISSED** [review]

File: src/main/java/com/softserve/rms/entities/Person.java [hunk 2]
```diff
@@ -44,7 +44,8 @@
 //    @JoinColumn(name = "role_id")
 //    private Role role;
 
-    @OneToMany(mappedBy = "user", orphanRemoval = true)
-    private List<ResourceTemplate> resource_templates = new ArrayList<>();
+    @OneToMany(mappedBy = "person", orphanRemoval = true)
+    private List<ResourceTemplate> resource_templates;
```
Comment (halinayatseniuk) [inline src/main/java/com/softserve/rms/entities/Person.java]: Change field name to resourceTemplates.

File: src/main/java/com/softserve/rms/entities/Person.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+public class Person {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+
+    @Column(name ="first_name", nullable = false, length = 50)
+    private String firstName;
+
+    @Column(name ="last_name", nullable = false, length = 50)
```
Comment (billyxd3) [inline src/main/java/com/softserve/rms/entities/Person.java]: remove `name ="last_name"`

File: src/main/java/com/softserve/rms/entities/Person.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+@ToString(
+     exclude = {"resource_templates"})
+
+public class Person {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+
+    @Column(name ="first_name", nullable = false, length = 50)
```
Comment (billyxd3) [inline src/main/java/com/softserve/rms/entities/Person.java]: remove `name="first_name"`

File: src/main/java/com/softserve/rms/entities/ResourceTemplate.java [hunk 5]
```diff
@@ -15,10 +15,11 @@
     private String name;
 
     private String description;
-    //TODO mapping on users table
 
-    private Long userId;
+    @ManyToOne(fetch = FetchType.EAGER)
+    @JoinColumn(name = "person_id")
```
Comment (billyxd3) [inline src/main/java/com/softserve/rms/entities/ResourceTemplate.java:20]: remove annotations `@JoinColumn (name ="person_id")`

**Review by halinayatseniuk — APPROVED** [review]

**Review by ArturSydor — APPROVED** [review]

**Review by billyxd3 — APPROVED** [review]

**Pull request merged**